runtime/rastrace/method_trace.c

```diff
@@ -477,10 +480,28 @@ traceMethodArgObject(J9VMThread *thr, UDATA* arg0EA, char* cursor, UDATA length)
 		J9Class* clazz = J9OBJECT_CLAZZ(thr, object);
 		J9ROMClass * romClass = clazz->romClass;
 		J9UTF8* className = J9ROMCLASS_CLASSNAME(romClass);
+		J9JavaVM *vm = thr->javaVM;
+
+		if (clazz == J9VMJAVALANGSTRING_OR_NULL(vm)) {
+			/* string arg */
+			char stringArgBuffer[DEFAULT_BUFFER_LENGTH];
 
-		/* TODO: handle arrays */
+			J9InternalVMFunctions const * const vmFuncs = thr->javaVM->internalVMFunctions;
+			char *stringArgUTF8 = vmFuncs->copyStringToUTF8WithMemAlloc(thr, object, J9_STR_NULL_TERMINATE_RESULT, "", 0, stringArgBuffer, DEFAULT_BUFFER_LENGTH, NULL);
+
+			if(DEFAULT_STRING_LENGTH < strlen(stringArgUTF8)) {
```

Comment:
In both cases, you dont need to print the class name since we already know its a string. You can just do `"(String)%.32s...`